src/config.rs

```diff
@@ -197,7 +203,8 @@ pub struct Config {
     mouse_bindings: Vec<MouseBinding>,
 
     /// Path to a shell program to run on startup
-    shell: Option<String>,
+    #[serde(default)]
+    shell: Option<Shell>,
```

Comment:
Will probably want to enforce that `Config` holds an owned `Shell` so this would gain a `'static` bound.